core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,114 @@
+package com.baeldung.charstreams;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.*;
+
+public class CharStreamsUnitTest {
+
+    private static DailyTodo TODO = new DailyTodo("Code", LocalDateTime.now(), true);
+    private static DailyTodo TODO_1 = new DailyTodo("Sleep", LocalDateTime.now(), false);
+    private static DailyTodo TODO_2 = new DailyTodo("Read", LocalDateTime.now(), false);
+    public static String TODO_LIST_PW_FILE = "src/main/resources/pw_todos.txt";
+    public static String TODO_LIST_FW_FILE = "src/main/resources/fw_todos.txt";
+    private static File pwFile = new File(TODO_LIST_PW_FILE); 
+    private static File fwFile =  new File(TODO_LIST_FW_FILE);
+
+    @BeforeEach
+    public void createFile() throws IOException {
+        try{
+            pwFile.mkdirs();
+            pwFile.createNewFile();
+            fwFile.mkdirs();
+            fwFile.createNewFile();

```

Comment:
Let's use the Files API instead here, and avoid the explicit creation of the new file. The `PrintWriter` and `FileWriter` will already take care of file creation for us.

```suggestion
        Files.createDirectories(Paths.get(TODO_LIST_PW_FILE).getParent());
        Files.createDirectories(Paths.get(TODO_LIST_FW_FILE).getParent());

```